modules/ml/src/test/java/org/apache/ignite/ml/naivebayes/gaussian/GaussianNaiveBayesTrainerTest.java

```diff
@@ -0,0 +1,171 @@
+        trainer = new GaussianNaiveBayesTrainer();
+    }
+
+    @Test
+    public void fit_LinearlySeparableData_retunsCorrectModel() {
+        Map<Integer, double[]> cacheMock = new HashMap<>();
+        for (int i = 0; i < twoLinearlySeparableClasses.length; i++)
+            cacheMock.put(i, twoLinearlySeparableClasses[i]);
+
+        GaussianNaiveBayesModel mdl = trainer.fit(
+            cacheMock,
+            parts,
+            (k, v) -> VectorUtils.of(Arrays.copyOfRange(v, 1, v.length)),
+            (k, v) -> v[0]
+        );
+
+        TestUtils.assertEquals(0, mdl.apply(VectorUtils.of(100, 10)), PRECISION);
+        TestUtils.assertEquals(1, mdl.apply(VectorUtils.of(10, 100)), PRECISION);
+    }
+
+    @Test
+    public void fit_returnsCorrectLabelProbalities() {
+
+        GaussianNaiveBayesModel model = trainer.fit(
+            new LocalDatasetBuilder<>(data, parts),
+            (k, v) -> VectorUtils.of(Arrays.copyOfRange(v, 0, v.length - 1)),
+            (k, v) -> v[2]
+        );
+
+        Assert.assertEquals(3. / data.size(), model.getClassProbabilities()[0], PRECISION);
+        Assert.assertEquals(2. / data.size(), model.getClassProbabilities()[1], PRECISION);
+    }
+
+    @Test
+    public void fit_setEquiprobableClasses_returnsEquivalentProbalities() {
+        GaussianNaiveBayesTrainer trainer = new GaussianNaiveBayesTrainer()
+            .withEquiprobableClasses();
+
+        GaussianNaiveBayesModel model = trainer.fit(
+            new LocalDatasetBuilder<>(data, parts),
+            (k, v) -> VectorUtils.of(Arrays.copyOfRange(v, 0, v.length - 1)),
+            (k, v) -> v[2]
+        );
+
+        Assert.assertEquals(.5, model.getClassProbabilities()[0], PRECISION);
+        Assert.assertEquals(.5, model.getClassProbabilities()[1], PRECISION);
+    }
+
+    @Test
+    public void fit_setPriorProbabilities_returnsPresetProbalities() {
+        double[] priorProbabilities = new double[] {.35, .65};
+        GaussianNaiveBayesTrainer trainer = new GaussianNaiveBayesTrainer()
+            .setPriorProbabilities(priorProbabilities);
+
+        GaussianNaiveBayesModel model = trainer.fit(
+            new LocalDatasetBuilder<>(data, parts),
+            (k, v) -> VectorUtils.of(Arrays.copyOfRange(v, 0, v.length - 1)),
+            (k, v) -> v[2]
+        );
+
+        Assert.assertEquals(priorProbabilities[0], model.getClassProbabilities()[0], PRECISION);
+        Assert.assertEquals(priorProbabilities[1], model.getClassProbabilities()[1], PRECISION);
+    }
+
+    @Test
+    public void fit_returnsCorrectMeans() {
+
+        GaussianNaiveBayesModel model = trainer.fit(
+            new LocalDatasetBuilder<>(singleLabeldata_1, parts),
+            (k, v) -> VectorUtils.of(Arrays.copyOfRange(v, 0, v.length - 1)),
+            (k, v) -> v[2]
+        );
+
+        Assert.assertArrayEquals(new double[] {2.0, 2. / 3.}, model.getMeans()[0], PRECISION);
+    }
+
+    @Test
+    public void fit_returnsCorrectVariances() {
```

Comment:
As I know, such naming is not legal in our codestyle
"testCorrectVariancesReturning" maybe

Reply:
I agree that `method_contidion_expectedResult` approach to name tests is not ideal. 
If it necessary I can rename all my tests, but I can’t find any reference about test methods naming in [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines#CodingGuidelines-Naming)
Could you say what exactly is not legal in my test? (I can suppose `_` or the tested method name) 

Also, the `test*` prefix in test methods names is also redundant, there is the annotation `@Test`.
As unit-tests are a service code (not a product code), can we retreat back from the code style to increase readability? I.e. underscore is used in examples [tutorial](https://github.com/apache/ignite/tree/master/examples/src/main/java/org/apache/ignite/examples/ml/tutorial)

As a compromise variant, I can suggest smth like `expectedResultWhenCondition`. (`returnsCorrectVariances`, `returnsEquivalentProbalitiesWhensetEquiprobableFlagSet`, etc.) (but anyway, even here underscore will increase readability)
What do you think?

Reply:
I renamed all my tests `test***` to pattern.